rs/nns/integration_tests/src/test_proposal_with_golden_nns_state.rs

```diff
@@ -0,0 +1,107 @@
+use std::env;
+
+use ic_base_types::PrincipalId;
+use ic_ledger_core::tokens::Tokens;
+use ic_nns_common::pb::v1::ProposalId;
+use ic_nns_test_utils::{
+    nns_canister_upgrade::NnsCanisterUpgrade,
+    state_test_helpers::{
+        adopt_proposal, nns_create_neuron_with_stake, nns_create_super_powerful_neuron,
+        nns_get_network_economics_parameters, nns_propose_upgrade_nns_canister,
+        nns_wait_for_proposal_execution, wait_for_canister_upgrade_to_succeed,
+    },
+};
+use ic_nns_test_utils_golden_nns_state::new_state_machine_with_golden_nns_state_or_panic;
+
+#[test]
+fn test_proposal_with_golden_nns_state() {
+    // Step 1: Prepare the world
+
+    // Make sure we can get a test version of NNS Governance. This enables adopting open proposals.
+    let mut nns_canister_upgrade = NnsCanisterUpgrade::new("governance-test");
+
+    // Step 1.1: Load golden nns state into a StateMachine.
+    let state_machine = new_state_machine_with_golden_nns_state_or_panic();
+
+    // Save initial state for the ultimate assertion of this test.
+    let network_economics = nns_get_network_economics_parameters(&state_machine);
```

Comment:
This could instead be hard-coded, right? I do not have a strong preference, esp since it seems like this test will not exist indefinitely.

Reply:
Closing as this parameter is used on all runs of this test target.